static/src/js/components/SavedProjects/SavedProjects.jsx

```diff
@@ -17,195 +17,184 @@ import Spinner from '../Spinner/Spinner'
-                    <th className="saved-projects-table__actions-heading">Actions</th>
-                  </tr>
-                </thead>
-                <tbody>
-                  {
-                    savedProjects.map((project) => {
-                      const {
-                        created_at: createdAt,
-                        id,
-                        name,
-                        path
-                      } = project
-
-                      const projectName = !name ? 'Untitled Project' : name
-
-                      const [pathname] = path.split('?')
-                      const sharePath = `${edscHost}${pathname}?projectId=${id}`
-                      const projectPath = this.projectTo(path, id)
-
-                      return (
-                        <tr key={id}>
-                          <td className="saved-projects-table__project-name">
-                            <PortalLinkContainer
-                              to={projectPath}
-                              onClick={() => { onChangePath(projectPath) }}
+  return (
+    <div className="saved-projects">
+      <h2 className="route-wrapper__page-heading">
+        Saved Projects
+      </h2>
+      {
+        (isLoading && !isLoaded) && (
+          <Spinner
+            className="saved-projects__spinner"
+            type="dots"
+            color="gray"
+            size="small"
+            role="status"
+          />
+        )
+      }
+      {
+        isLoaded && (
+          projects.length > 0 ? (
+            <Table className="saved-projects-table">
+              <thead>
+                <tr>
+                  <th className="saved-projects-table__project-name-heading">Project Name</th>
+                  <th className="saved-projects-table__contents-heading">Contents</th>
+                  <th className="saved-projects-table__created-heading">Created</th>
+                  <th className="saved-projects-table__actions-heading">Actions</th>
+                </tr>
+              </thead>
+              <tbody>
+                {
+                  projects.map((project) => {
+                    const {
+                      created_at: createdAt,
+                      id,
+                      name,
+                      path: projectPathValue
+                    } = project
+
+                    const projectName = !name ? 'Untitled Project' : name
+
+                    const [pathname] = projectPathValue.split('?')
+                    const sharePath = `${edscHost}${pathname}?projectId=${id}`
+                    const finalProjectPath = projectTo(projectPathValue, id)
+
+                    return (
+                      <tr key={id}>
+                        <td className="saved-projects-table__project-name">
+                          <PortalLinkContainer
+                            to={finalProjectPath}
+                            onClick={() => { onChangePath(finalProjectPath) }}
```

Comment:
These onClick methods could be `useCallback`s to prevent them causing the component to rerender